test/e2e/autoscaling/cluster_size_autoscaling.go

```diff
@@ -1226,7 +1226,8 @@ func deleteNodePool(name string) {
 
 func getPoolNodes(f *framework.Framework, poolName string) []*v1.Node {
 	nodes := make([]*v1.Node, 0, 1)
-	nodeList := framework.GetReadyNodesIncludingTaintedOrDie(f.ClientSet)
+	nodeList, err := e2enode.GetReadyNodesIncludingTaintedOrDie(f.ClientSet)
+	framework.ExpectNoError(err)
```

Comment:
This is an interesting change. `framework.ExpectNoError()` is essentially a wrapper for `ExpectNoErrorWithOffset()` which uses a [`gomega.ExpectWithOffset()`](https://godoc.org/github.com/onsi/gomega#ExpectWithOffset)
> This is most useful in helper functions that make assertions. If you want Gomega's error message to refer to the calling line in the test (as opposed to the line in the helper function) set the first argument of `ExpectWithOffset` appropriately.

The comments in `ExpectNoErrorWithOffset()` give a better example of this:
https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/util.go#L1370-L1371

Originally `e2enode.GetReadyNodesIncludingTaintedOrDie` makes the call to `gomega.ExpectWithOffset(1, err).()` which i reckon means that the error won't be logged as coming from `e2enode.GetReadyNodesIncludingTaintedOrDie` but from `getPoolNodes` in this case.
If we now make `e2enode.GetReadyNodesIncludingTaintedOrDie` return the error and `getPoolNodes` return the error, then error will be logged not by `getPoolNodes` but by the method that calls `getPoolNodes`.

My thinking is that we should keep the call to `gomega.ExpectWithOffset` in `e2enode.GetReadyNodesIncludingTaintedOrDie` and log the error in `getPoolNodes` otherwise we are changing the behavior of the tests and possibly affecting people's workflow.
Is this appropriate behavior?

wdyt?

I also mentioned this in one of my PRs https://github.com/kubernetes/kubernetes/pull/79753#discussion_r301208958
/cc @andrewsykim @timothysc

Reply:
need more input from people

Reply:
+1 to @alejandrox1 's case, I'd say switch it back

Reply:
moved this convo onto top level comments https://github.com/kubernetes/kubernetes/pull/79879#issuecomment-511402267